variables.tf

```diff
@@ -0,0 +1,153 @@
+
+######################################
+# HPCS Instance Variables
+######################################
+variable "is_hpcs_instance_exist" {
+  type        = bool
+  description = "Determines to if HPCS instance exists or not"
+  default     = false
+}
+variable "failover_units" {
+  type        = number
+  description = "Number of failover units"
+  default     = null
+}
+variable "location" {
+  default     = "us-south"
+  type        = string
+  description = "Location of HPCS Instance. Allowed values are us-south and us-east"
+  validation {
+    condition     = var.location == "us-south" || var.location == "us-east"
+    error_message = "Hpcs initialisation is only supported in us-south and us-east via terraform."
+  }
+}
+variable "service_name" {
```

Comment:
Please add regex validation, e.g:

variable "cluster_name" {
  description = "Name of the cluster"
  type        = string
  validation  {
      error_message = "Cluster name must begin and end with a letter and contain only letters, numbers, and - characters."
      condition     = can(regex("^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$", var.cluster_name))
    }
}